app/Http/Controllers/Api/RestoreDatabaseController.php

```
@@ -0,0 +1,333 @@
+
+            return response()->json([
+                'message' => 'Database restore initiated successfully.',
+                'restore_job_id' => $restoreJobId,
+            ], 200);
+        } catch (\Exception $e) {
+            return response()->json([
+                'message' => 'Failed to initiate restore.',
+                'error' => $e->getMessage(),
+            ], 500);
+        }
+    }
```

PHPMD (2.15.0)

The method restore() has a Cyclomatic Complexity of 15. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)